### PR TITLE
Fix overallocate by factor of four -- this was len * sizeof(guint8*) instead of len * sizeof(guint8), which devolves here to just alloca(len).

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build OS X i386

File: mono/mini/mini-x86.c [hunk 1]
```diff
@@ -597,7 +597,7 @@ mono_arch_get_argument_info (MonoMethodSignature *csig, int param_count, MonoJit
 
 	/* Avoid g_malloc as it is not signal safe */
 	len = sizeof (CallInfo) + (sizeof (ArgInfo) * (csig->param_count + 1));
-	cinfo = (CallInfo*)g_newa (guint8*, len);
+	cinfo = (CallInfo*)alloca (len);
 	memset (cinfo, 0, len);
```
Comment (vargaz) [inline mono/mini/mini-x86.c]: This should use g_alloca ().

**Comment by jaykrell** [comment]
Oh `g_alloca`? I read `g_newa`. Let's see. `g_newa` uses `alloca` directly though.

**Comment by jaykrell** [comment]
`g_alloca` is just `alloca`. What's the point?
`#define g_alloca(size)		alloca (size)`

**Comment by jaykrell** [comment]
We use alloca directly roughly twice as often as g_alloca.
```
cd mono
$ git grep g_alloca . | wc -l
      24
$ git grep \)alloca . | wc -l
      43
```

**Comment by jaykrell** [comment]
Better? I submitted a separate PR to change many `alloca` to `g_newa`.

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64 Interpreter

**Pull request merged**